CODEOWNERS

```diff
@@ -1 +1,2 @@
-* @dotnet/templating-engine-maintainers
\ No newline at end of file
+* @dotnet/templating-engine-maintainers
+docs/ @KathleenDollard
```

Comment:
@bozturkMSFT 
- can we also add template-engine-maintainers here? We need two reviews from them as well as per current settings. 
- should it be '/docs/?

Also I think it's nice to make the following changes to the rules
- in case doc owner haven't approved the PR, merging is blocked. 
However, open for debate here.

Reply:
I do not think we need the last rule. We still may have changes that do not need PM approval. For example, adding a template to the "Available templates" wiki page. There is no need to block merging in this case.

Reply:
Added the maintainers team to the owners of docs.
Updated to `/docs/` as the previous value meant "any docs folder under any directory".
I had already enabled requiring a review from code owners. For changes that don't need PM approval, we can just override the approval requirement since almost all the team has admin rights.